Add detail to PostgreSQL doc (#23716)
sourcegraph/sourcegraph: doc/admin/external_services/postgres.md

You can use your own PostgreSQL v12+ server with Sourcegraph if you wish. For example, you may prefer this if you already have existing backup infrastructure around your own PostgreSQL server, wish to use Amazon RDS, etc.

## General recommendations

> If you choose to set up your own PostgreSQL server, please note **we do not recommend using the same PostgreSQL instance for different databases**. Although it is technically possible to point both databases at the same PostgreSQL instance, we suggest either:

Reviewer: It's technically possible _now_ but we also make no guarantee that some minor update may cause table names to conflict if they're pointed at the same instance.

It's a **requirement** for them to be different _databases_ and **strongly recommended** for them to be difference servers and/or hosts.

> If you choose to set up your own PostgreSQL server, please note **we do not recommend using the same PostgreSQL instance for different databases**. Although it is technically possible to point both databases at the same PostgreSQL instance, we suggest either:

Author: ```suggestion
If you choose to set up your own PostgreSQL server, please note **we strongly recommended for each database to be set up in different servers and/or hosts**. We suggest either:
```

Author: @efritz is that clearer? Feel free to edit
